homeassistant/components/livebox/config_flow.py

```diff
@@ -0,0 +1,164 @@
+        self.data_schema[vol.Optional(CONF_PORT, default=DEFAULT_PORT)] = str
+        self.data_schema[vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME)] = str
+        self.data_schema[vol.Required(CONF_PASSWORD)] = str
+
+        return self.async_show_form(
+            step_id="init", data_schema=vol.Schema(self.data_schema)
+        )
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow start."""
+
+        if user_input is not None:
+            self.host = user_input[CONF_HOST]
+            self.port = user_input[CONF_PORT]
+            self.username = user_input[CONF_USERNAME]
+            self.password = user_input[CONF_PASSWORD]
+            return await self.async_step_link()
+
+        return self.async_show_form(
+            step_id="init", data_schema=vol.Schema(self.data_schema)
+        )
+
+    async def async_step_link(self, user_input=None):
+        """Step for link router."""
+
+        errors = {}
+
+        from aiosysbus import Sysbus
+
+        try:
+            box = Sysbus()
+            await box.open(
+                host=self.host,
+                port=self.port,
+                username=self.username,
+                password=self.password,
+            )
+            return await self._entry_from_box(box)
+
+        except AuthenticationRequired:
+            errors["base"] = "register_failed"
+
+        except CannotConnect:
+            LOGGER.error("Error connecting to the Livebox at %s", self.host)
+            errors["base"] = "linking"
+
+        except Exception:  # pylint: disable=broad-except
+            LOGGER.exception("Unknown error connecting with Livebox at %s", self.host)
+            errors["base"] = "linking"
+
+        # If there was no user input, do not show the errors.
+        if user_input is None:
+            errors = {}
+
+        return self.async_show_form(step_id="link", errors=errors)
+
+    async def _entry_from_box(self, box):
+        """Return a config entry from an initialized box."""
+        config = await box.system.get_deviceinfo()
+        box_id = config["status"]["SerialNumber"]
+
+        # Remove all other entries of hubs with same ID or host
+        same_hub_entries = [
+            entry.entry_id
+            for entry in self.hass.config_entries.async_entries(DOMAIN)
+            if entry.data["box_id"] == box_id
+        ]
+
+        if same_hub_entries:
+            await asyncio.wait(
+                [
+                    self.hass.config_entries.async_remove(entry_id)
+                    for entry_id in same_hub_entries
+                ]
+            )
+
+        return self.async_create_entry(
+            title=TEMPLATE_SENSOR.format(""),
+            data={
+                "box_id": box_id,
+                "host": self.host,
+                "port": self.port,
+                "username": self.username,
+                "password": self.password,
+            },
+        )
+
```

Comment:
There's no import step. Please add an import step as an entry point when importing the config yaml to the config flow.